src/ecrc-frontend/src/components/page/success/Success.test.js

```diff
@@ -85,4 +118,49 @@ describe("Success Page Component", () => {
     );
     expect(failedPayment.toJSON()).toMatchSnapshot();
   });
+
+  test("Validate Retry payment", async () => {
+    const history = createMemoryHistory();
+    const { container } = render(
+      <Router history={history}>
+        <MemoryRouter initialEntries={[failureUrl]}>
+          <Success page={page} />
+        </MemoryRouter>
+      </Router>
+    );
+
+    fireEvent.click(getByText(container, "Click here to try again"));
+
+    await wait(() => {
+      expect(saveApplicationInfo).toHaveBeenCalled();
+    });
+  });
+
+  test("Validate Print", () => {
+    const history = createMemoryHistory();
+    const { container } = render(
+      <Router history={history}>
+        <MemoryRouter initialEntries={[succesUrl]}>
+          <Success page={page} />
+        </MemoryRouter>
+      </Router>
+    );
+
+    fireEvent.click(getByText(container, "Print"));
+    expect(window.print).toHaveBeenCalled();
+  });
+
+  test("Validate Download", () => {
+    const history = createMemoryHistory();
+    const { container } = render(
+      <Router history={history}>
```

Comment:
do we need both `Router` and `MemoryRouter`?

Reply:
Not 100% sure, can check to see if Router can be used in place of MemoryRouter for the snapshots. I think Route was needed because we need access to the history outside of the routing block.

Reply:
removed router and history as these tests are not checking the history

Reply:
Note: Router doesnt take initialEntries and MemoryRouter doesnt take a history param